### PR TITLE
Mobile access restriction GA setting form-not-saving-on-submit bug fix

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

This GA setting for mobile access restrictions has already been released and successfully migrated from the feature flag. There was only an error with domains that have certain names (like those that have a '.' in their name) that prevent the setting from being changed. The privacy form would look like it was submitted but the setting would not update properly. This PR will fix that change so that the GA setting can be safely relied upon.

## Technical Summary

[This is the main related PR](https://github.com/dimagi/commcare-hq/pull/30783). The checkbox for the GA setting was not saving in certain domains. After investigation, we learned it was because `StaticToggle.enabled()` was being passed a domain object instead of a domain name. When it was passed an object, instead of throwing an error the `enabled` function checked the domain's `hr_name` (human-readable name).

In addition to the one-line fix, this PR also includes a change to the privacy form save method's argument to use the terminology `domain_obj` instead of `domain`, ~~and adding a `soft_assert` check to `StaticToggle.enabled()` that it is passed a string. The `soft_assert` email will be sent to the general ops listserv~~. The check that `StaticToggle.enabled()` is passed a string will be a separate PR because it has proven to be a more complicated fix (tests are failing due to `soft_assert` falling, `StaticToggle.enabled()` is passed non-string values in other parts of the codebase).

## Feature Flag

RESTRICT_MOBILE_ACCESS

### Safety story

Tested locally with a domain name with a `.` in it to make sure the bug fix works as intended. The soft_assert email output also looks correct.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/toggles.py [hunk 1]
```diff
@@ -156,6 +157,19 @@ def enabled(self, item, namespace=Ellipsis):
         ):
             # Don't even bother looking it up in the cache
             return False
+
+        # item should be a string
+        _soft_assert = soft_assert(notify_admins=False, send_to_ops=True)
```
Comment (esoergel) [inline corehq/toggles.py:162]: This part needs to be moved outside the function - usually to the top of the file.  That way `soft_assert` is only called once, when the code is first loaded.  That's how it tracks the number of hits for the exponential backoff.

Comment (AddisonDunn) [reply corehq/toggles.py:162]: Ah, gotcha. Fix coming.

**Review by esoergel — APPROVED** [review]

**Comment by esoergel** [comment]
Nitpick, you can use `git rebase -i` to drop commits from your branch.  Something like `git rebase -i HEAD~3` will let you play with just the past 3 commits without changing the base commit and screwing up existing reviews in github, as might happen with a rebase off `origin/master`.  I did a write-up on some of that here: https://github.com/dimagi/open-source/blob/es/writing-prs/Writing_PRs.md#iteractive-rebasing

**Comment by AddisonDunn** [comment]
> Nitpick, you can use `git rebase -i` to drop commits from your branch. Something like `git rebase -i HEAD~3` will let you play with just the past 3 commits without changing the base commit and screwing up existing reviews in github, as might happen with a rebase off `origin/master`. I did a write-up on some of that here: https://github.com/dimagi/open-source/blob/es/writing-prs/Writing_PRs.md#iteractive-rebasing

Ah and dropping the commits via a rebase will be much cleaner than adding an awkward "revert" commit.

**Pull request merged**